plugin/docker/container-log.go

```diff
@@ -28,28 +29,56 @@ func (clf *containerLogFile) isTty(ctx context.Context) bool {
 	return true
 }
 
-func (clf *containerLogFile) Open(ctx context.Context) (plugin.SizedReader, error) {
-	opts := types.ContainerLogsOptions{ShowStdout: true, ShowStderr: true}
-	rdr, err := clf.client.ContainerLogs(ctx, clf.containerName, opts)
+func (clf *containerLogFile) cachedContent(ctx context.Context) ([]byte, error) {
+	cachedContent, err := plugin.CachedOp("Content", clf, 15*time.Second, func() (interface{}, error) {
+		opts := types.ContainerLogsOptions{ShowStdout: true, ShowStderr: true}
+		rdr, err := clf.client.ContainerLogs(ctx, clf.containerName, opts)
+		if err != nil {
+			return nil, err
+		}
+
+		var buf bytes.Buffer
+		var n int64
+		if clf.isTty(ctx) {
+			if n, err = buf.ReadFrom(rdr); err != nil {
+				return nil, err
+			}
+		} else {
+			// Write stdout and stderr to the same buffer.
+			if n, err = stdcopy.StdCopy(&buf, &buf, rdr); err != nil {
+				return nil, err
+			}
+		}
+		journal.Record(ctx, "Read %v bytes of %v log", n, clf.containerName)
+
+		return buf.Bytes(), nil
+	})
+
 	if err != nil {
 		return nil, err
 	}
 
-	var buf bytes.Buffer
-	var n int64
-	if clf.isTty(ctx) {
-		if n, err = buf.ReadFrom(rdr); err != nil {
-			return nil, err
-		}
-	} else {
-		// Write stdout and stderr to the same buffer.
-		if n, err = stdcopy.StdCopy(&buf, &buf, rdr); err != nil {
-			return nil, err
-		}
+	return cachedContent.([]byte), nil
+}
+
+func (clf *containerLogFile) Metadata(ctx context.Context) (plugin.EntryMetadata, error) {
```

Comment:
So this essentially gets rid of the default behavior of using `Open` to determine the size implicitly?

Reply:
Also, why add a separate method rather than using CachedOpen directly?

Reply:
Oh, good idea. I'll update.